src/tools/opt-dist/src/environment/windows.rs

```diff
@@ -0,0 +1,54 @@
+use crate::environment::Environment;
+use crate::utils::io::move_directory;
+use camino::Utf8PathBuf;
+use std::io::{Cursor, Read};
+use zip::ZipArchive;
+
+pub(super) struct WindowsEnvironment {
+    checkout_dir: Utf8PathBuf,
+}
+
+impl WindowsEnvironment {
+    pub fn new() -> Self {
+        Self { checkout_dir: std::env::current_dir().unwrap().try_into().unwrap() }
+    }
+}
+
+impl Environment for WindowsEnvironment {
+    fn checkout_path(&self) -> Utf8PathBuf {
+        self.checkout_dir.clone()
+    }
+
+    fn downloaded_llvm_dir(&self) -> Utf8PathBuf {
+        self.checkout_path().join("citools/clang-rust")
+    }
+
+    fn opt_artifacts(&self) -> Utf8PathBuf {
+        self.checkout_path().join("opt-artifacts")
+    }
+
+    fn build_root(&self) -> Utf8PathBuf {
+        self.checkout_path()
+    }
+
+    fn prepare_rustc_perf(&self) -> anyhow::Result<()> {
+        const PERF_COMMIT: &str = "9dfaa35193154b690922347ee1141a06ec87a199";
+
+        let url = format!("https://github.com/rust-lang/rustc-perf/archive/{PERF_COMMIT}.zip");
+        let mut response = Vec::new();
+        ureq::get(&url).call()?.into_reader().read_to_end(&mut response)?;
```

Comment:
let's use curl/PowerShell for this for consistency with bootstrap; I don't want to add a rust networking stack into our dependency tree. you can move some of the code in download.rs in build_helper if that makes it easier.

Reply:
Ah, I was so proud of getting rid of creating a temporary file (which was done in Python) and instead downloading and directly unzipping it :( :D

Reply:
You could curl to a pipe to retain that property

Reply:
Yeah but I'd really like to avoid completing the circle and not go from bash to Python to Rust and then back to bash :sweat_smile: Since this tool will mostly only be used on CI, we discussed with jyn that this dependency should be OK.

Reply:
curl and pipes don't need any bash. `std::process::Command::new("curl").args(...)`

Reply:
That's true, it doesn't look that bad. Still, if I can avoid using `curl` manually, that would be nice, ultimately I'd like to download/upload stuff from/to S3 here and that would be a bit cumbersome with just `curl`.